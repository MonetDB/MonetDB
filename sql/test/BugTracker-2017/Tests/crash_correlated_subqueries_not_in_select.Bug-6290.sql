SELECT s.name AS schema_nm, s.system AS is_system_schema
, (SELECT CAST(COUNT(*) as int) FROM sys.functions f WHERE f.schema_id = s.id AND f.type <> 2 AND f.id NOT IN (SELECT function_id FROM sys.systemfunctions)) AS "# user defined functions"
, (SELECT CAST(COUNT(*) as int) FROM sys.functions f WHERE f.schema_id = s.id AND f.type = 2 AND f.id NOT IN (SELECT function_id FROM sys.systemfunctions)) AS "# user defined procedures"
 FROM sys.schemas AS s
WHERE s.name IN ('json', 'profiler', 'sys');

