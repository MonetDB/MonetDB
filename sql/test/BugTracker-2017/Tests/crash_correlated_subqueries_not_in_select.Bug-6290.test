statement ok
START TRANSACTION

statement ok
CREATE TABLE "myschemas" ("id" INTEGER, "name" VARCHAR(1024), "authorization" INTEGER, "owner" INTEGER, "system" BOOLEAN)

statement ok rowcount 7
INSERT INTO "myschemas" VALUES (2000, 'sys', 2, 3, true),(2114, 'tmp', 2, 3, true),(7446, 'json', 3, 3, true),(7542, 'profiler', 3, 3, true),(7738, 'wlc', 3, 3, true),(7757, 'wlr', 3, 3, true),(8270, 'logging', 3, 3, true)

query TIII rowsort
SELECT s.name AS schema_nm, s.system AS is_system_schema
, (SELECT CAST(COUNT(*) as int) FROM sys.functions f WHERE f.schema_id = s.id AND f.type <> 2 AND f.id NOT IN (SELECT sf.id as function_id FROM sys.functions sf WHERE sf.system)) AS "user defined functions count"
, (SELECT CAST(COUNT(*) as int) FROM sys.functions f WHERE f.schema_id = s.id AND f.type = 2 AND f.id NOT IN (SELECT sf.id as function_id FROM sys.functions sf WHERE sf.system)) AS "user defined procedures count"
 FROM sys.myschemas AS s
WHERE s.name IN ('json', 'profiler', 'sys')
----
json
1
0
0
profiler
1
0
0
sys
1
0
0

query TIII rowsort
SELECT s.name AS schema_nm, s.system AS is_system_schema
, (SELECT CAST(COUNT(*) as int) FROM sys.functions f WHERE f.schema_id = s.id AND f.type <> 2 AND NOT f.system) AS "user defined functions count"
, (SELECT CAST(COUNT(*) as int) FROM sys.functions f WHERE f.schema_id = s.id AND f.type = 2 AND NOT f.system) AS "user defined procedures count"
 FROM sys.myschemas AS s
WHERE s.name IN ('json', 'profiler', 'sys')
----
json
1
0
0
profiler
1
0
0
sys
1
0
0

query I rowsort
SELECT (SELECT 1 FROM functions f WHERE f.schema_id = s.id AND f.id IN (SELECT -1 FROM functions x)) FROM myschemas AS s
----
NULL
NULL
NULL
NULL
NULL
NULL
NULL

query I rowsort
SELECT (SELECT 1 FROM functions f WHERE f.schema_id = s.id AND f.id NOT IN (SELECT x.id FROM functions x)) FROM myschemas AS s
----
NULL
NULL
NULL
NULL
NULL
NULL
NULL

statement ok
ROLLBACK
