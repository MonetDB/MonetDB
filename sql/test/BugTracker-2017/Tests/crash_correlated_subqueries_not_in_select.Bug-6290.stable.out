stdout of test 'crash_correlated_subqueries_not_in_select.Bug-6290` in directory 'sql/test/BugTracker-2017` itself:


# 13:40:17 >  
# 13:40:17 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33807" "--set" "mapi_usock=/var/tmp/mtest-32606/.s.monetdb.33807" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch2/dinther/INSTALL/var/MonetDB/mTests_sql_test_BugTracker-2017" "--set" "embedded_r=yes" "--set" "embedded_py=true"
# 13:40:17 >  

# MonetDB 5 server v11.25.22
# This is an unreleased version
# Serving database 'mTests_sql_test_BugTracker-2017', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 128bit integers
# Found 15.589 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2017 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://toulouse.da.cwi.nl:33807/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-32606/.s.monetdb.33807
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/Python module loaded
# MonetDB/R   module loaded

# SQL catalog created, loading sql scripts once
# loading sql script: 09_like.sql
# loading sql script: 10_math.sql
# loading sql script: 11_times.sql
# loading sql script: 12_url.sql
# loading sql script: 13_date.sql
# loading sql script: 14_inet.sql
# loading sql script: 15_querylog.sql
# loading sql script: 16_tracelog.sql
# loading sql script: 17_temporal.sql
# loading sql script: 18_index.sql
# loading sql script: 20_vacuum.sql
# loading sql script: 21_dependency_functions.sql
# loading sql script: 22_clients.sql
# loading sql script: 23_skyserver.sql
# loading sql script: 25_debug.sql
# loading sql script: 26_sysmon.sql
# loading sql script: 27_rejects.sql
# loading sql script: 39_analytics.sql
# loading sql script: 39_analytics_hge.sql
# loading sql script: 40_geom.sql
# loading sql script: 40_json.sql
# loading sql script: 40_json_hge.sql
# loading sql script: 41_md5sum.sql
# loading sql script: 45_uuid.sql
# loading sql script: 46_gsl.sql
# loading sql script: 46_profiler.sql
# loading sql script: 51_sys_schema_extension.sql
# loading sql script: 72_fits.sql
# loading sql script: 74_netcdf.sql
# loading sql script: 75_lidar.sql
# loading sql script: 75_shp.sql
# loading sql script: 75_storagemodel.sql
# loading sql script: 80_statistics.sql
# loading sql script: 80_udf.sql
# loading sql script: 80_udf_hge.sql
# loading sql script: 85_bam.sql
# loading sql script: 90_generator.sql
# loading sql script: 90_generator_hge.sql
# loading sql script: 99_system.sql

# 13:40:17 >  
# 13:40:17 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-32606" "--port=33807"
# 13:40:17 >  

#SELECT s.name AS schema_nm, s.system AS is_system_schema
#, (SELECT CAST(COUNT(*) as int) FROM sys.functions f WHERE f.schema_id = s.id AND f.type <> 2 AND f.id NOT IN (SELECT function_id FROM sys.systemfunctions)) AS "# user defined functions"
#, (SELECT CAST(COUNT(*) as int) FROM sys.functions f WHERE f.schema_id = s.id AND f.type = 2 AND f.id NOT IN (SELECT function_id FROM sys.systemfunctions)) AS "# user defined procedures"
# FROM sys.schemas AS s
#WHERE s.name IN ('json', 'profiler', 'sys');
% .,	.,	.,	. # table_name
% schema_nm,	is_system_schema,	"# user defined functions",	"# user defined procedures" # name
% varchar,	boolean,	int,	int # type
% 8,	5,	1,	1 # length
[ "sys",	true,	0,	0	]
[ "json",	true,	0,	0	]
[ "profiler",	true,	0,	0	]

# 13:40:18 >  
# 13:40:18 >  "Done."
# 13:40:18 >  

